packages/belos/tpetra/example/BlockGmres/BlockGmresTpetraExFile.cpp

```diff
@@ -0,0 +1,261 @@
+
+  using SCT = typename Teuchos::ScalarTraits<ST>;
+  using MT  = typename SCT::magnitudeType;
+  using MV  = typename Tpetra::MultiVector<ST,LO,GO,NT>;
+  using OP  = typename Tpetra::Operator<ST,LO,GO,NT>;
+
+  using tmap_t       = Tpetra::Map<LO,GO,NT>;
+  using tvector_t    = Tpetra::Vector<ST,LO,GO,NT>;
+  using tcrsmatrix_t = Tpetra::CrsMatrix<ST,LO,GO,NT>;
+
+  using MVT = typename Belos::MultiVecTraits<ST,MV>;
+  using OPT = typename Belos::OperatorTraits<ST,MV,OP>;
+
+  using Teuchos::RCP;
+  using Teuchos::rcp;
+  using Teuchos::ParameterList;
+
+  Teuchos::GlobalMPISession mpiSession (&argc, &argv, &std::cout);
+  const auto Comm = Tpetra::getDefaultComm();
+  const int MyPID = Comm->getRank();
+
+  bool verbose = false;
+  bool success = true;
+
+  try {
+    bool proc_verbose = false;
+    bool debug = false;
+    int frequency = -1;        // frequency of status test output.
+    int blocksize = 1;         // blocksize
+    int numrhs = 1;            // number of right-hand sides to solve for
+    int maxiters = -1;         // maximum number of iterations allowed per linear system
+    int maxsubspace = 50;      // maximum number of blocks the solver can use for the subspace
+    int maxrestarts = 15;      // number of restarts allowed
+    std::string filename("orsirr1.hb");
+    MT tol = 1.0e-5;           // relative residual tolerance
+    std::string ortho = "ICGS";// orthogonalization method
+
+    Teuchos::CommandLineProcessor cmdp(false,true);
+    cmdp.setOption("verbose","quiet",&verbose,"Print messages and results.");
+    cmdp.setOption("debug","nondebug",&debug,"Print debugging information from solver.");
+    cmdp.setOption("frequency",&frequency,"Solvers frequency for printing residuals (#iters).");
+    cmdp.setOption("filename",&filename,"Filename for test matrix.  Acceptable file extensions: *.hb,*.mtx,*.triU,*.triS");
+    cmdp.setOption("tol",&tol,"Relative residual tolerance used by GMRES solver.");
+    cmdp.setOption("num-rhs",&numrhs,"Number of right-hand sides to be solved for.");
+    cmdp.setOption("block-size",&blocksize,"Block size used by GMRES.");
+    cmdp.setOption("max-iters",&maxiters,"Maximum number of iterations per linear system (-1 = adapted to problem/block size).");
+    cmdp.setOption("max-subspace",&maxsubspace,"Maximum number of blocks the solver can use for the subspace.");
+    cmdp.setOption("max-restarts",&maxrestarts,"Maximum number of restarts allowed for GMRES solver.");
+    cmdp.setOption("ortho",&ortho,"Orthogonalization being used by GMRES solver.");
+    if (cmdp.parse(argc,argv) != Teuchos::CommandLineProcessor::PARSE_SUCCESSFUL) {
+      return -1;
+    }
+    if (!verbose)
+      frequency = -1;  // reset frequency if test is not verbose
+
+    proc_verbose = ( verbose && (MyPID==0) );
+
+    if (proc_verbose) {
+      std::cout << Belos::Belos_Version() << std::endl << std::endl;
+    }
+    //
+    // Get the problem
+    //
```

Comment:
Take a style of comments and keep it (`//`seems great)

Reply:
Resolved